tests/optimizer/test_parameter_flattener.py

```diff
@@ -2,7 +2,7 @@
 
 import pytest
 
-from tensorwaves.optimizer.minuit import ParameterFlattener
+from tensorwaves.optimizer._parameter import ParameterFlattener
```

Comment:
Good that `ParameterFlattener` is hidden now. I guess `get_backend_modules` and `gradient_creator` can also be hidden, but let's do that in a separate PR.